src/device/mod.rs

```diff
@@ -336,13 +339,24 @@ struct Runtime {
 }
 
 impl Device {
+    fn check_ipv6_availability() -> bool {
```

Comment:
Oh there is no need to base this on the availability (fallback mechanisms are designed for that). Therefore it is enough to base it on `self.features.ipv6`